source/grid/manifold_lib.cc

```diff
@@ -178,6 +152,78 @@ SphericalManifold<dim,spacedim>::push_forward_gradient(const Point<spacedim> &sp
 }
 
 // ============================================================
+// SphericalManifold
+// ============================================================
+
+template <int dim, int spacedim>
+SphericalManifold<dim,spacedim>::SphericalManifold(const Point<spacedim> center):
+  center(center)
+{}
+
+template <int dim, int spacedim>
+Point<spacedim>
+SphericalManifold<dim,spacedim>::
+get_new_point (const Point<spacedim> &P1, const Point<spacedim> &P2, double w) const
+{
+  if ( P1 == P2 )
+    return P1;
+
+  if (dim == spacedim)
+    {
+      return Point<spacedim>((1-w) * P1 + w * P2);
+    }
+  else
+    {
+      Assert(w >=0.0 && w <= 1.0,
+             ExcMessage("w should be in the range [0.0,1.0]."));
+      Assert((center-P1).norm() == (center-P2).norm(),
```

Comment:
you can't compare floating point numbers for equality. You need to compare them in relative terms:
```
  Assert (std::fabs((center-P1).norm() - (center-P2).norm()) < 1e-12 * ((center-P1).norm() + (center-P2).norm()),
```